invenio_rdm_records/resources/serializers.py

```diff
@@ -9,32 +9,85 @@
 """Record response serializers."""
 
 import json
+from functools import partial
 
 from flask_resources.serializers import JSONSerializer
 
+from invenio_rdm_records.vocabularies import Vocabularies
+
 
 class UIJSONSerializer(JSONSerializer):
     """UI JSON serializer implementation."""
 
     def _serialize_access_right(self, obj):
-        """Dump ui config for `access_right` field."""
+        """Inject ui config for `access_right` field."""
         return dict(
             access_right=dict(
                 category=obj["access"]["access_right"]
             ))
 
+    def _serialize_ui_options_from_vocabulary(
```

Comment:
This function works for the currently present Vocabularies but maybe it's worth considering a recursive approach if we expect depth>2 in any future ones.

Reply:
ping @lnielsen